src/logger/filters.ts

```diff
@@ -0,0 +1,90 @@
+/*
```

Comment:
I refactored out the "filters" from the original logger into its own function,
and then refactored it to speed things up a bit.

When the perf tests run, they log a LOT through to the fs.  That becomes a stream bottleneck, and backpressure eventually makes its way from fs => transport stream => worker  thread => logger => main process.  improving this redactor perf should keep the logger fast.

![Screenshot 2023-06-29 at 2 33 50 PM](https://github.com/forcedotcom/sfdx-core/assets/4261788/5e6994e8-3d33-45a0-96e7-02805dd58f1f)